### DISCUSSION TITLE
2024/01/28/Linux-service-operation-notes-01/

### DISCUSSION THREAD

**Discussion by giscus** [cyclinbox/cyclinbox.github.io#13] [General]
# 2024/01/28/Linux-service-operation-notes-01/

如题。 原先的服务器快到期了，因此这周初新购进一台华为云的服务器。在配置服务器的时候，顺带温习了一下这些知识点。

https://wz.anoms.top/2024/01/28/Linux-service-operation-notes-01/

**Comment by Hazelandinstant** [comment]
crond这个感觉很牛！求细讲！

**Reply by cyclinbox** [reply]
root用户下使用crond添加定时任务的方法大概就是文章中提到的这些。

非root用户的话，记住一个指令 `crontab` 即可，这个指令可以控制单个用户下的定时任务。

使用方法：

+ `crontab -l`  ： 列出当前用户的定时任务。
+ `crontab -e` ： 编辑当前用户的定时任务列表。运行这个指令会进入 `/var/spool/cron/<username>` 文件的编辑界面（虽然此时界面上显示可能是`/tmp/`下面的一个临时文件），然后在此添加定时任务即可。

需要注意的是，使用`crontab -e`编辑的定时任务只能以当前用户的身份进行执行，因此cron文件的语法也略有不同，具体而言就是少了第六列的用户名。仍然以文章中那个早上9点整的爬虫脚本为例，如果我们以普通用户的身份通过`crontab -e`添加定时任务的话，那么文件内容大概是这样（`#`开头的行全部是注释）：

```text
# Example of job definition:
# .---------------- minute (0 - 59)
# |  .------------- hour (0 - 23)
# |  |  .---------- day of month (1 - 31)
# |  |  |  .------- month (1 - 12) OR jan,feb,mar,apr ...
# |  |  |  |  .---- day of week (0 - 6) (Sunday=0 or 7) OR sun,mon,tue,wed,thu,fri,sat
# |  |  |  |  |
# *  *  *  *  * command to be executed
0 9 * * * /home/warrenz/spyder.py
```

要检查任务是否执行成功，除了文章中提到的方法外，好像也没什么更有效的方法（我太菜了😢）。不过，有问题欢迎交流~
